pkg/operator/rgw/controller.go

```diff
@@ -101,3 +106,35 @@ func (c *ObjectStoreController) onDelete(obj interface{}) {
 		logger.Errorf("failed to delete object store %s. %+v", objectStore.Name, err)
 	}
 }
+
+func storeChanged(oldStore, newStore ObjectStoreSpec) bool {
+	if oldStore.DataPool.Replicated.Size != newStore.DataPool.Replicated.Size {
```

Comment:
does this correctly cover the case where the data pool was using erasure coding but now it's using replication?

Reply:
since it's not supported to change the pool type, we could either raise that error here or let ceph return an error when we try to set the replication. To keep the crd change detection simple, it seemed appropriate to defer to ceph for the error.